syft/generic/abstract/sendable.py

```diff
@@ -1,11 +1,56 @@
+# from abc import abstractmethod
+from typing import TYPE_CHECKING
+from typing import Union
+
+import syft as sy
 from syft.serde.syft_serializable import SyftSerializable
 from syft.generic.abstract.object import AbstractObject
 
+# from syft.generic.abstract.pointer import AbstractPointer
+from syft.workers.abstract import AbstractWorker
+
+# this if statement avoids circular imports between base.py and pointer.py
+if TYPE_CHECKING:
+    from syft.generic.abstract.pointer import AbstractPointer
+
 
 class AbstractSendable(AbstractObject, SyftSerializable):
     """
     This layers functionality for sending objects between workers on top of AbstractObject.
     """
 
-    def send(self, destination):
-        return self.owner.send_obj(self, destination)
+    def send(
+        self, destination: Union[AbstractWorker, str], garbage_collect_data=None
+    ) -> "AbstractPointer":
+        ptr_id = sy.ID_PROVIDER.pop()
+
+        destination = self.owner.get_worker(destination)
+
+        pointer = self.create_pointer(
+            location=destination,
+            id_at_location=self.id,
+            owner=self.owner,
+            ptr_id=ptr_id,
+            garbage_collect_data=garbage_collect_data,
+        )
+
+        self.owner.send_obj(self, destination)
+
+        return pointer
+
+    # TODO: Make the `create_pointer` method below an abstract method. We can't do this yet, because
+    # we're relying on the wrapper tensors to provide the functionality of creating pointers, which
+    # means that many of the custom tensor types don't yet have pointer types or ways to create
+    # pointers.
+
+    # @abstractmethod
+    def create_pointer(
+        self,
+        location: AbstractWorker = None,
+        id_at_location: (str or int) = None,
+        owner: AbstractWorker = None,
+        ptr_id: (str or int) = None,
+        garbage_collect_data: bool = True,
+        **kwargs,
+    ) -> "AbstractPointer":
+        pass
```

Comment:
I think it is a good idea to raise `NotImplementedException` here. This is useful for debugging whenever we try to send an object that does not override this method.

Reply:
Yeah, that might be a reasonable compromise. Making it an `abstractmethod` raises an error immediately when it detects that a child class hasn't implemented it, which is too strict for an incremental rework. Raising `NotImplemented` here is a good way to split the difference in the mean time. 👍

Reply:
I think your forgot to add the `NotImplementedException` ?